src/common.rs

```diff
@@ -0,0 +1,20 @@
+pub(crate) const DEFAULT_ARCH: &str = "x86_64";
+use anyhow::{ensure, Context, Result};
+use tokio::process::Command;
+
+/// Run a `tokio::process::Command` and return a `Result` letting us know whether or not it worked.
+pub(crate) async fn exec(cmd: &mut Command) -> Result<()> {
+    let output = cmd
+        .output()
+        .await
+        .context(format!("Unable to start command '{:?}'", cmd))?;
```

Comment:
It may be worth emitting stdout/stderr at DEBUG level. I've used a few tools that eat subprocess outputs, and it can be pretty obnoxious if you need to debug things.

Reply:
> It may be worth emitting stdout/stderr at DEBUG level. I've used a few tools that eat subprocess outputs, and it can be pretty obnoxious if you need to debug things.

Good idea:

DEBUG

```
$ cargo run -- --log-level debug build variant
[2023-06-27T21:20:17Z DEBUG twoliter::common] Running: Command { std: DOCKER_BUILDKIT="1" "docker" "build" "--file" "/tmp/.tmpasrafv/Twoliter.dockerfile" "--tag" "twoliter:latest" "--build-arg=BASE=public.ecr.aws/bottlerocket/bottlerocket-sdk-x86_64:v0.33.0" "/tmp/.tmpasrafv/context", kill_on_drop: false }
[+] Building 0.9s (11/11) FINISHED
 => [internal] load build definition from Twoliter.dockerfile                                                                            0.0s
 => => transferring dockerfile: 163B                                                                                                     0.0s
 => [internal] load .dockerignore                                                                                                        0.0s
 => => transferring context: 2B                                                                                                          0.0s
 => resolve image config for docker.io/docker/dockerfile:1.4.3                                                                           0.6s
 => CACHED docker-image://docker.io/docker/dockerfile:1.4.3@sha256:9ba7531bd80fb0a858632727cf7a112fbfd19b17e94c4e84ced81e24ef1a0dbc      0.0s
 => [internal] load .dockerignore                                                                                                        0.0s
 => [internal] load build definition from Twoliter.dockerfile                                                                            0.0s
 => [internal] load metadata for public.ecr.aws/bottlerocket/bottlerocket-sdk-x86_64:v0.33.0                                             0.0s
 => [1/2] FROM public.ecr.aws/bottlerocket/bottlerocket-sdk-x86_64:v0.33.0                                                               0.0s
 => [internal] load build context                                                                                                        0.0s
 => => transferring context: 62B                                                                                                         0.0s
 => CACHED [2/2] COPY --chmod=755 buildsys /usr/local/bin                                                                                0.0s
 => exporting to image                                                                                                                   0.0s
 => => exporting layers                                                                                                                  0.0s
 => => writing image sha256:51559d41d0729d4d22eaae579634235e24c896503ba9fccdd2de4c1a9e5a7039                                             0.0s
 => => naming to docker.io/library/twoliter:latest       
```

INFO

```
cargo run -- --log-level info build variant
```